packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -344,9 +426,24 @@ async function startWatcher(
       pageFiles.clear()
       devPageFiles.clear()
 
-      const sortedKnownFiles: string[] = [...knownFiles.keys()].sort(
-        sortByPageExts(nextConfig.pageExtensions)
-      )
+      const sortedKnownFiles: string[] = [...knownFiles.keys()].sort((a, b) => {
+        // First, prioritize regular routes over parallel routes
+        const aHasParallel = a.includes('/@')
```

Comment:
better to have accurate assertion as well, also path need to normalize because windows doesn't have `/`